src/pythium/histogramming/Managers.py

```diff
@@ -0,0 +1,317 @@
+from pythium.common.tools import h5py_to_ak, json_to_ak, parquet_to_ak
+from pythium.histogramming.objects  import Observable, NTupSyst, TreeSyst, WeightSyst, CrossProduct
+from pythium.histogramming.binning import _Binning
+from glob import glob 
+import dask
+import os
+from collections import defaultdict
+from pprint import pprint
+import boost_histogram as bh
+import time 
+import awkward as ak
+from pythium.common.functor import Functor
+import numpy as np
+import re
+from pythium.common.logger import ColoredLogger
+
+logger = ColoredLogger()
+
+class _InputManager(object):
+    def __init__(self, xps, cfg):
+        read_methods = {
+                        'parquet': 'ak_parquet',
+                        'json': 'ak_json',
+                        'root': 'uproot',
+                        'h5': 'ak_h5'
+                        }
+        self.xps = xps
+        self.indirs = cfg["general"]["indir"]
+        self.from_pyth = cfg["general"]["frompythium"]
+        self.ext = cfg["general"]["informat"]
+        self.sample_sel = cfg["general"]["samplesel"]
+        self.reader = read_methods[self.ext]
+    
+    def required_variables(self):
+        req_vars: List[Observable] = []
+        xp_to_req = defaultdict(list)
+        for xp in self.xps:
+            sample, region, obs, systematic, template = xp
+            required_variables = []
+            obs_vars, _region_sel_vars, sample_sel_vars, syst_vars = [],[],[],[]
+        
+            obs_vars = [Observable(reqvar, reqvar, obs.binning, obs.dataset) for reqvar in obs.builder.req_vars if reqvar not in [xp["observable"].name for xp in self.xps if xp["observable"].builder.new] ]
```

Comment:
add weights to obs_vars

`obs_vars.extend([Observable(obs.weights, obs.weights, obs.binning, obs.dataset)])` (line 43)

Reply:
Note: check required that weights passed as  str or list when adding to required variables.
If set obs.weights = 1.2 in config, get the following error 
`beartype.roar.BeartypeCallHintParamViolation: @beartyped Observable.__init__() parameter var=1.2 violates type hint typing.Union[str, typing.List[str]], as 1.2 not str or list`

Reply:
weights should now be properly supported in case of providing a string, float or list/ndarray of floats